Appz/Appz/Apps/Yelp.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  Yelp.swift
+//  Pods
+//
+//  Created by Mariam AlJamea on 12/29/15.
+//  Copyright © 2015 kitz. All rights reserved.
+//
+
+public extension Applications {
+    
+    public struct Yelp: ExternalApplication {
+        
+        public typealias ActionType = Applications.Yelp.Action
+        
+        public let scheme = "yelp4:"
+        public let fallbackURL = "https://m.yelp.com/"
+        
+        public init() {}
+    }
+}
+
+// MARK: - Actions
+// Taken from: https://www.yelp.com/developers/documentation/v2/iphone
+
+public extension Applications.Yelp {
+    
+    public enum Action {
+        case Open
+        case Search(query: String)
+        case SearchLocation(query: String, loc: String)
+        case SearchCategory(cat: String)
+        case SearchCatLoc(loc: String, cat: String)
+        case Business(id: String)
+    }
+}
+
+extension Applications.Yelp.Action: ExternalApplicationAction {
+    
+    public var paths: ActionPaths {
+        
+        switch self {
+        case .Open:
+            return ActionPaths(
+                app: Path(
+                    pathComponents: ["app"],
+                    queryParameters: [:]
+                ),
+                web: Path()
+            )
+            
+        case .Search(let query):
+            return ActionPaths(
+                app: Path(
+                    pathComponents: ["", "search"],
+                    queryParameters: ["terms": query]
+                ),
+                web: Path(
+                    pathComponents: ["search"],
+                    queryParameters: ["terms": query]
+                )
+            )
+            
+        case .SearchLocation(let query, let loc):
+            return ActionPaths(
+                app: Path(
+                    pathComponents: ["", "search"],
```

Comment:
@Mazyod Good morning, I checked this URL: http://www.yelp.com/biz/yelp-san-francisco taken from https://www.yelp.com/developers/documentation/v2/phone_search then I tried it with and without empty String and it gave me same result so I didn't add it. So do you think I need to add it?